File: ObjectiveCExample/NotificationExtesnionServices/NotificationService.m
#import "NotificationService.h"
 
@interface NotificationService ()
 
@property (nonatomic, strong) void (^contentHandler)(UNNotificationContent *contentToDeliver);
@property (nonatomic, strong) UNMutableNotificationContent *bestAttemptContent;
@property (nonatomic, strong) UNNotificationRequest *receivedRequest;
 
@end
 
@implementation NotificationService
 
- (void)didReceiveNotificationRequest:(UNNotificationRequest *)request withContentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler {
    self.contentHandler = contentHandler;
    self.receivedRequest = request;
    self.bestAttemptContent = [request.content mutableCopy];
    if (self.bestAttemptContent != nil)
    {
        [iZooto didReceiveNotificationExtensionRequestWithBundleName:@"com.iZooto.ObjectiveCExample" request:self.receivedRequest bestAttemptContent:self.bestAttemptContent contentHandler: self.contentHandler];
    }
}
 
- (void)serviceExtensionTimeWillExpire {
    // Called just before the extension will be terminated by the system.
    // Use this as an opportunity to deliver your "best attempt" at modified content, otherwise the original push payload will be used.
    self.contentHandler(self.bestAttemptContent);
}
 
@end

File: iZootoiOSProject/iZootoNotificationExtendsServices/NotificationService.swift
//
//  NotificationService.swift
//  iZootoNotificationExtendsServices
//
//  Created by AMIT_SDK_DEVELOPER on 23/01/23.
//  Copyright © 2023 Amit. All rights reserved.
//

import UserNotifications
import iZootoiOSSDK
class NotificationService: UNNotificationServiceExtension {
    var contentHandler: ((UNNotificationContent) -> Void)?
    var bestAttemptContent: UNMutableNotificationContent?
    var receivedRequest: UNNotificationRequest!

    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
           self.receivedRequest = request;
           self.contentHandler = contentHandler
           bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent)
           if let bestAttemptContent = bestAttemptContent {
               iZooto.didReceiveNotificationExtensionRequest(bundleName :"com.iZootoiOSProject", soundName: "",request: receivedRequest, bestAttemptContent: bestAttemptContent,contentHandler: contentHandler)
              
       }


       }
       override func serviceExtensionTimeWillExpire() {
           if let contentHandler = contentHandler, let bestAttemptContent =  bestAttemptContent {
               iZooto.didReceiveNotificationExtensionRequest(bundleName :"com.iZootoiOSProject", soundName: "",request: receivedRequest, bestAttemptContent: bestAttemptContent,contentHandler: contentHandler)
           }
       }

}
